src/m365/spo/commands/commandset/commandset-set.ts

```diff
@@ -0,0 +1,225 @@
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        if (args.options.listType && SpoCommandsetSetCommand.listTypes.indexOf(args.options.listType) < 0) {
+          return `${args.options.listType} is not a valid list type. Allowed values are ${SpoCommandsetSetCommand.listTypes.join(', ')}`;
+        }
+
+        if (args.options.scope && SpoCommandsetSetCommand.scopes.indexOf(args.options.scope) < 0) {
+          return `${args.options.scope} is not a valid scope. Allowed values are ${SpoCommandsetSetCommand.scopes.join(', ')}`;
+        }
+
+        if (args.options.location && SpoCommandsetSetCommand.locations.indexOf(args.options.location) < 0) {
+          return `${args.options.location} is not a valid location. Allowed values are ${SpoCommandsetSetCommand.locations.join(', ')}`;
+        }
+
+        if (!args.options.newTitle && !args.options.listType && !args.options.clientSideComponentProperties && !args.options.location) {
+          return `Please specify option to be updated`;
+        }
+
+        return validation.isValidSharePointUrl(args.options.webUrl);
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['id', 'title', 'clientSideComponentId'] }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Updating ListView Command Set ${args.options.clientSideComponentId} to site '${args.options.webUrl}'...`);
+    }
+
+    if (!args.options.scope) {
+      args.options.scope = 'Site';
+    }
+
+    const location: string = this.getLocation(args.options.location ? args.options.location : '');
+
+    try {
+      const requestBody: any = {
+      };
+
+      if (args.options.newTitle) {
+        requestBody.Title = args.options.newTitle;
+      }
+
+      if (args.options.location) {
+        requestBody.Location = location;
+      }
+
+      if (args.options.listType) {
+        requestBody.RegistrationId = args.options.listType === 'List' ? "100" : "101";
+      }
+
+      if (args.options.clientSideComponentProperties) {
+        requestBody.ClientSideComponentProperties = args.options.clientSideComponentProperties;
+      }
+
+      const id = await this.getCommandSetId(args.options);
+
+      const requestOptions: any = {
+        url: `${args.options.webUrl}/_api/${args.options.scope}/UserCustomActions('${formatting.encodeQueryParameter(id)}')`,
+        headers: {
+          accept: 'application/json;odata=nometadata',
+          'X-HTTP-Method': 'MERGE'
+        },
+        data: requestBody,
+        responseType: 'json'
+      };
+
+      await request.post<CustomAction>(requestOptions);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getCommandSetId(options: Options): Promise<string> {
```

Comment:
In general, if we have `return` statement in `if` block it makes no sense to have an `else` or `else if` block. 
Let's consider the following example
```
if(A) {
return A;
} 
else if(B) {
return B;
} 
else {
return C;
}
```

This will be actually the same as
```
if(A) {
return A;
} 

if(B) {
return B;
} 

return C;
```

since we have `return`. The result and logic is the same but we avoid the necessary `else` which makes the code cleaner. 

What do you think? Maybe we should adjust this method accordingly?

Reply:
You're totally right, that last else is not needed since we have a return in the other if statements. Good catch!

Reply:
TBH this comment was not only about last else but if we have either `return` or `throw` in any of the `else` or `else if` statements then we may skip all that and just have `if` .
So in this case we may also (I think) avoid the `esle if` and just have `if`